src/Raven.Client/Http/RequestExecutor.cs

```diff
@@ -1342,6 +1447,83 @@ public static async Task<Stream> ReadAsStreamUncompressedAsync(HttpResponseMessa
             return true;
         }
 
+        private bool ShouldBroadcast<TResult>(RavenCommand<TResult> command)
+        {
+            if (command is IBroadcast == false)
+                return false;
+
+            if (TopologyNodes == null || 
+                TopologyNodes.Count < 2)
+                return false;
+
+            return true;
+        }
+
+        private async Task<TResult> Broadcast<TResult>(RavenCommand<TResult> command, SessionInfo sessionInfo, CancellationToken token)
+        {
+            var broadcastCommand = command as IBroadcast;
+            if (broadcastCommand == null)
+                throw new InvalidOperationException("You can broadcast only commands that implement 'IBroadcast'.");
+
+            command.FailedNodes = new Dictionary<ServerNode, Exception>(); // clear the current failures
+
+            using (var broadcastCts = CancellationTokenSource.CreateLinkedTokenSource(token))
+            {
+                var tasks = SendToAllNodes<TResult>(sessionInfo, broadcastCommand, broadcastCts.Token);
+
+                return await WaitForResult(command, tasks, broadcastCts).ConfigureAwait(false);
+            }
+        }
+
+        private async Task<TResult> WaitForResult<TResult>(RavenCommand<TResult> command, Dictionary<Task, (RavenCommand<TResult> Command, int Index)> tasks, CancellationTokenSource broadcastCts)
+        {
+            while (tasks.Count > 0)
+            {
+                var completed = await Task.WhenAny(tasks.Keys).ConfigureAwait(false);
+                if (completed.IsCanceled || completed.IsFaulted)
+                {
+                    var failed = tasks[completed];
+                    var node = _nodeSelector.Topology.Nodes[failed.Index];
+                    command.FailedNodes[node] = completed.Exception.ExtractSingleInnerException();
```

Comment:
If task is canceled I'm not 100% sure if Exception is set?

Reply:
done